cider-velox/test/CiderOperatorTest.cpp

```diff
@@ -325,27 +324,159 @@ TEST_F(CiderOperatorTest, avg_on_col_cider) {
+
+TEST_F(CiderOperatorTest, avg_on_col_null_nogroupby) {
+  RowVectorPtr vector =
+      makeRowVector({makeFlatVector<int64_t>(
+                         {2499109626526694126, 2342493223442167775, 4077358421272316858}),
+                     makeFlatVector<int32_t>({581869302, -708632711, -133711905}),
+                     makeFlatVector<double>(
+                         {0.90579193414549275, 0.96886777112423139, 0.63235925003444637}),
+                     makeFlatVector<bool>({true, false, false}),
+                     makeFlatVector<int32_t>(3, nullptr, nullEvery(1))});
+
+  createDuckDbTable({vector});
+  auto veloxPlan = PlanBuilder()
+                       .values({vector})
+                       .filter("c2 < 24.0")
+                       .project({"c4"})
+                       .partialAggregation({}, {"avg(c4) as avg_price"}, {})
+                       .finalAggregation()
+                       .planNode();
+
+  auto resultPtr = CiderVeloxPluginCtx::transformVeloxPlan(veloxPlan);
+  auto duckdbSql = "SELECT avg(c4) as avg_price FROM tmp WHERE c2 < 24.0 ";
+
+  assertQuery(veloxPlan, duckdbSql);
+  // TODO : (ZhangJie) Enable this after Yizhong fix the null parsing in  cider
+  // (https://jira.devtools.intel.com/browse/POAE7-2342). Now, expected results are null,
+  // while we get -Infinity/Infinity.
+  GTEST_SKIP();
+  assertQuery(resultPtr, duckdbSql);
+}
+
+TEST_F(CiderOperatorTest, partial_avg) {
+  auto data = makeRowVector({makeFlatVector<int64_t>(10, [](auto row) { return row; })});
+  createDuckDbTable({data});
+  auto veloxPlan = PlanBuilder()
+                       .values({data})
+                       .project({"c0"})
+                       .partialAggregation({}, {"avg(c0) as avg_ccccc"}, {})
+                       .planNode();
+
+  auto resultPtr = CiderVeloxPluginCtx::transformVeloxPlan(veloxPlan);
+
+  auto duckdbSql = "SELECT row(45, 10)";
+
+  assertQuery(veloxPlan, duckdbSql);
+  assertQuery(resultPtr, duckdbSql);
+
+  const ::substrait::Plan substraitPlan = ::substrait::Plan();
+  auto expectedPlan =
+      PlanBuilder()
+          .values({data})
+          .addNode([&](std::string id, std::shared_ptr<const core::PlanNode> input) {
+            return std::make_shared<facebook::velox::plugin::CiderPlanNode>(
+                CiderPlanNode(id, {input}, input->outputType(), substraitPlan));
+          })
+          .planNode();
+  EXPECT_TRUE(PlanTansformerTestUtil::comparePlanSequence(resultPtr, expectedPlan));
+}
+
+TEST_F(CiderOperatorTest, partial_avg_null) {
+  auto data = makeRowVector({makeAllNullFlatVector<int32_t>(3)});
+
+  createDuckDbTable({data});
+
+  auto veloxPlan = PlanBuilder()
+                       .values({data})
+                       .project({"c0"})
+                       .partialAggregation({}, {"avg(c0) as avg_ccccc"}, {})
+                       .planNode();
+
+  auto resultPtr = CiderVeloxPluginCtx::transformVeloxPlan(veloxPlan);
+
+  auto duckdbSql = "SELECT row(null, 3)";
```

Comment:
why it's 3 here.
`null, null, null` -> [null, 3]
`null, null, null, 4`-> [4,4] or [4,1]?

Reply:
the return value is approximate to null value (-9223372036854775807). it's a double type here or something else?

Reply:
> why it's 3 here. `null, null, null` -> [null, 3] `null, null, null, 4`-> [4,4] or [4,1]?

`null, null, null` -> null  `null, null, null, 4`-> [1,1] 

will modify duckdb sql and add new simple test.

Reply:
> the return value is approximate to null value (-9223372036854775807). it's a double type here or something else?

I think it's a double type.

Reply:
`-9223372036854775807` should not be correct as I don't consider null value when try casting bigint null to double null at cider side.